Extend current destruction docs (#103)
pluralsh/documentation: pages/operations/uninstall.md

---
title: Destroying the Cluster and Installations
description: How do I bring things down safely?
---

## Overview

Plural provides options to uninstall specific applications, tear down your clusters, and wipe any references to installations to start from zero. Uninstalling any Plural application or an entire Plural installation is a one-liner in your terminal or Cloud Shell. If you want to delete your Plural installation, make sure to run `plural destroy` before deleting your Git repository. If you delete your Git repository first, you will have to manually clean up all the resources that Plural has provisioned for you.&#x20;

## Purging the Cloud Shell

> If you'd like to move your Cloud Shell instance locally, you can sync your shell and delete the Cloud Shell instance from our servers. To sync your shell and delete your current Cloud Shell instance, use:

Author: @michaeljguarino is there any other reason you'd want to purge your cloud shell? Like starting over from a partial setup?

> If you'd like to move your Cloud Shell instance locally, you can sync your shell and delete the Cloud Shell instance from our servers. To sync your shell and delete your current Cloud Shell instance, use:

Reviewer: ```suggestion
If you created a Plural installation in our Cloud Shell and want to move it to your local dev environment, you can sync your shell and delete the Cloud Shell instance from our servers. To sync your shell and delete your current Cloud Shell instance, use:
```

 
 ```
 plural shell sync
 plural shell purge
 ```
 

>   The purge command will destroy your current Cloud Shell instance, but preserve your existing cluster and installations. Your account will still be pinned to the same cloud provider chosen at the beginning of your onboarding.

Reviewer: nit

```suggestion
The purge command will destroy your current Cloud Shell instance, but preserve your existing cluster and installations. Your account will still be pinned to the same cloud provider chosen at the beginning of your onboarding.
```

{% callout severity="danger" %}
Only do this if you're absolutely sure you want to bring down all associated resources with this repository.
{% /callout %}

## Terraform Destroy

To tear down the current cluster but leave installation references as pointers to the helm/terraform, cd into `bootstrap/terraform` and run:

```
terraform destroy
```

## Start from Zero

To remove all installation references in our API, run:

```
plural repos reset
```

This command does not interact with any infrastructure, but removes references to all installations. This will also reset any association with a specific cloud provider.

> To tear down a cluster and fully start over, run `plural destroy` and then `plural repos reset`.

Author: @michaeljguarino does a user need to do plural repos reset after plural destroy to remove those installations, or is that cleaned up now?

{% callout severity="danger" %}
Only do this if you're absolutely sure you want to bring down all associated resources with this repository.
{% /callout %}

> By default, previously installed applications will still appear in app.plural.sh after running `plural destroy`. To remove all installation history, run `plural repos reset` as documented below.

Reviewer: ```suggestion
By default, previously installed applications will still appear in your [installed applications](https://app.plural.sh/installed) page on app.plural.sh after running `plural destroy`. To remove all installation history, run `plural repos reset` as documented below.
```

## Terraform Destroy

> To tear down the current cluster but leave installation references as pointers to the helm/terraform, cd into `bootstrap/terraform` and run:

Reviewer: ```suggestion
To tear down the current cluster but leave installation references as pointers to the Helm/Terraform, cd into `bootstrap/terraform` and run:
```

```
terraform destroy
```

## Start from Zero

> To remove all installation references in our API, run:

Reviewer: Is it clear what a "reference" is? It might be, just checking.

```
plural repos reset
```

This command does not interact with any infrastructure, but removes references to all installations. This will also reset any association with a specific cloud provider.

> To tear down a cluster and fully start over, run `plural destroy` and then `plural repos reset`.

Reviewer: Is it worth having a subheading that says "Fully Start Over" or "Destroy Everything" for these 2 commands together?
